lib/hubspot/contact_properties.rb

```diff
@@ -74,27 +84,45 @@ def filter_results(results, key, include, exclude)
       end
 
       def valid_property_params(params)
-        names             = %w(name groupName description fieldType formField type displayOrder label options)
-        valid_field_types = %w(textarea select text date file number radio checkbox)
-        valid_types       = %w(string number bool datetime enumeration)
-
-        params.each do |key, val|
-          return false unless names.include?(key)
-          return false if key == 'fieldType' && !valid_field_types.include?(val)
-          return false if key == 'type' && !valid_types.include?(val)
-        end
-        true
+        params.select { |key, val|
+          if PROPERTY_SPECS[:field_names].include?(key)
+            case key
+              when 'fieldType'
+                check_field_type(val)
+              when 'type'
+                check_type(val)
+              else
+                true
+            end
+          end
+        }
```

Comment:
This will silently drop fields as long as there is one valid field in the params, which is different from how it worked before. Is that the desired behavior?

(The same thing is true of the `valid_group_params` method).